src/overlays/actors/ovl_Boss_07/z_boss_07.c

```diff
@@ -242,500 +636,7211 @@ ActorProfile Boss_07_Profile = {
+    MAJORAS_WRATH_BODYPART_LEFT_SHIN             // MAJORAS_WRATH_BODYPART_LEFT_FOOT
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_07/func_80A06500.s")
+static u8 sShadowSizes[MAJORAS_WRATH_BODYPART_MAX] = {
+    MAJORAS_WRATH_SHADOW_SIZE_SMALL,       // MAJORAS_WRATH_BODYPART_HEAD
+    MAJORAS_WRATH_SHADOW_SIZE_EXTRA_LARGE, // MAJORAS_WRATH_BODYPART_TORSO
+    MAJORAS_WRATH_SHADOW_SIZE_EXTRA_LARGE, // MAJORAS_WRATH_BODYPART_PELVIS
+    MAJORAS_WRATH_SHADOW_SIZE_LARGE,       // MAJORAS_WRATH_BODYPART_LEFT_UPPER_ARM
+    MAJORAS_WRATH_SHADOW_SIZE_SMALL,       // MAJORAS_WRATH_BODYPART_LEFT_LOWER_ARM_ROOT
+    MAJORAS_WRATH_SHADOW_SIZE_SMALL,       // MAJORAS_WRATH_BODYPART_LEFT_FOREARM
+    MAJORAS_WRATH_SHADOW_SIZE_LARGE,       // MAJORAS_WRATH_BODYPART_RIGHT_UPPER_ARM
+    MAJORAS_WRATH_SHADOW_SIZE_SMALL,       // MAJORAS_WRATH_BODYPART_RIGHT_LOWER_ARM_ROOT
+    MAJORAS_WRATH_SHADOW_SIZE_SMALL,       // MAJORAS_WRATH_BODYPART_RIGHT_FOREARM
+    MAJORAS_WRATH_SHADOW_SIZE_LARGE,       // MAJORAS_WRATH_BODYPART_RIGHT_THIGH
+    MAJORAS_WRATH_SHADOW_SIZE_MEDIUM,      // MAJORAS_WRATH_BODYPART_RIGHT_SHIN
+    MAJORAS_WRATH_SHADOW_SIZE_SMALL,       // MAJORAS_WRATH_BODYPART_RIGHT_FOOT
+    MAJORAS_WRATH_SHADOW_SIZE_LARGE,       // MAJORAS_WRATH_BODYPART_LEFT_THIGH
+    MAJORAS_WRATH_SHADOW_SIZE_MEDIUM,      // MAJORAS_WRATH_BODYPART_LEFT_SHIN
+    MAJORAS_WRATH_SHADOW_SIZE_SMALL,       // MAJORAS_WRATH_BODYPART_LEFT_FOOT
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_07/func_80A06990.s")
+void Boss07_Wrath_FillShadowTex(Boss07* this, u8* tex, f32 weight) {
+    s32 index;
+    s32 i;
+    s32 baseX;
+    s32 baseY;
+    s32 x;
+    s32 y = 0;
+    s32 addY;
+    Vec3f lerp;
+    Vec3f pos;
+    Vec3f startVec;
+
+    for (i = 0; i < MAJORAS_WRATH_BODYPART_MAX; i++) {
+        // TODO: match with a continue
+        if ((weight == 0.0f) || ((y = sParentShadowBodyParts[i]) > BODYPART_NONE)) {
+            if (weight > 0.0f) {
+                VEC3F_LERPIMPDST(&lerp, &this->bodyPartsPos[i], &this->bodyPartsPos[y], weight);
+
+                pos.x = lerp.x - this->actor.world.pos.x;
+                pos.y = lerp.y - this->actor.world.pos.y + 76.0f + 30.0f + 30.0f + 100.0f;
+                pos.z = lerp.z - this->actor.world.pos.z;
+            } else {
+                pos.x = this->bodyPartsPos[i].x - this->actor.world.pos.x;
+                pos.y = this->bodyPartsPos[i].y - this->actor.world.pos.y + 76.0f + 30.0f + 30.0f + 100.0f;
+                pos.z = this->bodyPartsPos[i].z - this->actor.world.pos.z;
+            }
+
+            Matrix_MultVec3f(&pos, &startVec);
+
+            startVec.x *= 0.2f;
+            startVec.y *= 0.2f;
+
+            baseX = (u16)(s32)(startVec.x + 32.0f);
+            baseY = (u16)((s32)startVec.y * 64);
+
+            if (sShadowSizes[i] == MAJORAS_WRATH_SHADOW_SIZE_EXTRA_LARGE) {
+                for (y = 0, addY = -0x180; y < ARRAY_COUNT(sShadowExtraLargeMap); y++, addY += 0x40) {
```

Comment:
It looks like a lot of constants around here that are written in hex should probably be in dec

Reply:
In this case, it's because every single other shadow function uses hexadecimal here. I updated them all in this commit: https://github.com/zeldaret/mm/pull/1753/commits/c923e5566a1f2809af55a07b7206c54aa830d8e6

If it turns out we actually prefer hex, I can just revert the commit.